rearrange_cars_luiza.py

```diff
@@ -0,0 +1,182 @@
+        self.current_car_order = init_car_order
+        self.final_car_order = final_car_order
+        self.empty_spot_position = self.current_car_order.index(0)
+        self.print_console = True
+        self.car_position_dict = {car:pos for pos, car in enumerate(init_car_order) \
+                                  if car} #search the index of car in O(1) via dict
+        self.min_num_steps_left = None
+
+    def print_one_step(self, from_idx):
+
+        '''Prints a move of the car from position #from_idx to empty spot
+        '''
+
+        move_str = 'move from {0} to {1}'.format(from_idx, self.empty_spot_position)
+        if self.print_console:
+            print move_str
+        else:
+            logging.info(move_str)
+
+    def create_log_file(self):
+
+        '''In case the number of moves is more than 100
+        writes the output to the logfile instead of the console
+        '''
+        N = len(self.initial_car_order)
+        logging_message = 'Number of cars to arrange is {0} > 1000, movements are logged'.format(N)
+        logging.warning(logging_message)
+        logging.info('Initial car order:')
+        logging.info(self.initial_car_order)
+        logging.info('Final car order:')
+        logging.info(self.final_car_order)
+        logging.info('Moves:')
+
+    def move_car_to_empty_slot(self, car_to_move_position):
+
+        '''Arranges the car with car_to_move_position index to the empty slot
+        Arguments:
+            car_to_move_position: the position from where the car is moved
+        Returns:
+            None
+        '''
+
+        car_to_move = self.current_car_order[car_to_move_position]
+        self.current_car_order[self.empty_spot_position] = car_to_move
+        self.current_car_order[car_to_move_position] = 0
+
+        self.car_position_dict[car_to_move] = self.empty_spot_position
+        self.empty_spot_position = car_to_move_position
+
+    def get_not_arranged_car_index(self):
+
+        '''Finds first car index not yet standing on the final destination
+        index should never reach the end since there should be at least
+        one car not yet arranged
+        '''
+
+        index = 0
+        while (self.current_car_order[index] == self.final_car_order[index]) or \
+                                        (index == self.empty_spot_position):
+            index += 1
+
+        return index
+
+    def rearrange_one_car(self):
+
+        '''Puts to the current empty space the car
+        that should be on this place according to the final arrangement
+        if current empty space position is different from the final order empty
+        space position
+
+        Otherwise picks up the random not arranged car and places it to the empty space
+        '''
+
+        if self.final_car_order[self.empty_spot_position] == 0:
+            car_to_move_position = self.get_not_arranged_car_index()
+            car_to_move = self.current_car_order[car_to_move_position]

```

Comment:
Could you plz explain why you randomly put a car into the empty space when 0 is where it should be at the `final_car_order`? If that's the case, perhaps you can just return since no move is needed?